### PR TITLE
[Trend Micro Deep Security] - add new commands

### PR DESCRIPTION
**Pull request by GuyAfik**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-8423)

## Description
- Added the following commands:
  - ***trendmicro-create-onceonly-scheduled-task***
  - ***trendmicro-list-scheduled-task***
  - ***trendmicro-delete-scheduled-task***

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/TrendMicroDeepSecurity/Integrations/TrendMicroDeepSecurity/README.md [hunk 1]
```diff
@@ -24040,3 +24040,163 @@ Reset the value of a certain default policy setting
...
+
+#### Base Command
+
+`trendmicro-list-scheduled-task`
+
+#### Input
+
+| **Argument Name** | **Description** | **Required** |
+| --- | --- | --- |
+| task_id | Retrieve a specific task-ID. | Optional |
```
Comment (ShirleyDenkberg) [inline Packs/TrendMicroDeepSecurity/Integrations/TrendMicroDeepSecurity/README.md:24056]:
```suggestion
| task_id | The ID of the task to retrieve. | Optional | 
```

File: Packs/TrendMicroDeepSecurity/Integrations/TrendMicroDeepSecurity/README.md [hunk 2]
```diff
@@ -24040,3 +24040,163 @@ Reset the value of a certain default policy setting
...
+| TrendMicro.ScheduledTask.type | String | The type of the scheduled task. | 
+| TrendMicro.ScheduledTask.scheduleDetails.timeZone | String | The timezone of the scheduled task. | 
+| TrendMicro.ScheduledTask.scheduleDetails.recurrenceType | String | The recurrence type of the scheduled task. | 
+| TrendMicro.ScheduledTask.scheduleDetails.onceOnlyScheduleParameters.startTime | Number | The start time of the scheduled task. | 
+| TrendMicro.ScheduledTask.enabled | Boolean | Whether the scheduled task is enabled. | 
+| TrendMicro.ScheduledTask.nextRunTime | Date | The next run time of the scheduled task. | 
+| TrendMicro.ScheduledTask.scanForMalwareTaskParameters.computerFilter.type | String | The type of the computer filter of the scheduled task. | 
+| TrendMicro.ScheduledTask.scanForMalwareTaskParameters.computerFilter.computerID | Number | The computer ID of the scheduled task. | 
+| TrendMicro.ScheduledTask.scanForMalwareTaskParameters.timeout | String | The timeout for the scheduled task. | 
+| TrendMicro.ScheduledTask.ID | Number | The ID of the scheudled task. |
```
Comment (ShirleyDenkberg) [inline Packs/TrendMicroDeepSecurity/Integrations/TrendMicroDeepSecurity/README.md:24072]:
```suggestion
| TrendMicro.ScheduledTask.ID | Number | The ID of the scheduled task. | 
```

File: Packs/TrendMicroDeepSecurity/Integrations/TrendMicroDeepSecurity/README.md [hunk 3]
```diff
@@ -24040,3 +24040,163 @@ Reset the value of a certain default policy setting
...
+
+#### Base Command
+
+`trendmicro-delete-scheduled-task`
+
+#### Input
+
+| **Argument Name** | **Description** | **Required** |
+| --- | --- | --- |
+| task_id | The task-ID to delete. | Required |
```
Comment (ShirleyDenkberg) [inline Packs/TrendMicroDeepSecurity/Integrations/TrendMicroDeepSecurity/README.md:24193]:
```suggestion
| task_id | The ID of the task to delete. | Required | 
```

File: Packs/TrendMicroDeepSecurity/Integrations/TrendMicroDeepSecurity/TrendMicroDeepSecurity.yml [hunk 4]
```diff
@@ -3959,7 +3959,118 @@ script:
...
+        description: The computer ID of the scheduled task.
+        type: Number
+      - contextPath: TrendMicro.ScheduledTask.scanForMalwareTaskParameters.timeout
+        description: The timeout for the scheduled task.
+        type: String
+      - contextPath: TrendMicro.ScheduledTask.ID
+        description: The ID of the scheduled task.
+        type: Number
+  - arguments:
+    - description: The task-ID to delete.
```
Comment (ShirleyDenkberg) [inline Packs/TrendMicroDeepSecurity/Integrations/TrendMicroDeepSecurity/TrendMicroDeepSecurity.yml:4027]:
```suggestion
    - description: The ID of the task to delete.
```

File: Packs/TrendMicroDeepSecurity/Integrations/TrendMicroDeepSecurity/TrendMicroDeepSecurity.yml [hunk 5]
```diff
@@ -3959,7 +3959,118 @@ script:
...
+      name: task_id
+      required: true
+    description: Deletes a scheduled task.
+    name: trendmicro-delete-scheduled-task
+    outputs: []
+  - arguments:
+    - description: Retrieve a specific task-ID.
```
Comment (ShirleyDenkberg) [inline Packs/TrendMicroDeepSecurity/Integrations/TrendMicroDeepSecurity/TrendMicroDeepSecurity.yml:4034]:
```suggestion
    - description: The ID of the task to retrieve.
```

File: Packs/TrendMicroDeepSecurity/Integrations/TrendMicroDeepSecurity/TrendMicroDeepSecurity.yml [hunk 6]
```diff
@@ -3959,7 +3959,118 @@ script:
...
+        description: The type of the computer filter of the scheduled task.
+        type: String
+      - contextPath: TrendMicro.ScheduledTask.scanForMalwareTaskParameters.computerFilter.computerID
+        description: The computer ID of the scheduled task.
+        type: Number
+      - contextPath: TrendMicro.ScheduledTask.scanForMalwareTaskParameters.timeout
+        description: The timeout for the scheduled task.
+        type: String
+      - contextPath: TrendMicro.ScheduledTask.ID
+        description: The ID of the scheudled task.
```
Comment (ShirleyDenkberg) [inline Packs/TrendMicroDeepSecurity/Integrations/TrendMicroDeepSecurity/TrendMicroDeepSecurity.yml:4071]:
```suggestion
        description: The ID of the scheduled task.
```

**Comment by ShirleyDenkberg** [comment]
@JudahSchwartz Doc review completed.

File: Packs/TrendMicroDeepSecurity/Integrations/TrendMicroDeepSecurity/TrendMicroDeepSecurity.py [hunk 7]
```diff
@@ -1736,6 +1778,101 @@ def create_policy_command(client: Client, name: str, overrides: bool, parent_id:
...
+                context_output,
+                headers=["ID", "name", "type", "enabled", "lastRunTime"],
+                headerTransform=pascalToSpace,
+                date_fields=["lastRunTime", "nextRunTime"],
+                removeNull=True
+            )
+        )
+
+    return CommandResults(
+        readable_output="There aren't any scheduled tasks available"
```
Comment (JudahSchwartz) [inline Packs/TrendMicroDeepSecurity/Integrations/TrendMicroDeepSecurity/TrendMicroDeepSecurity.py:1872]:
```suggestion
        readable_output="There are no existing scheduled tasks."
```

**Review by JudahSchwartz — APPROVED** [review]
amazing, please tag doc review

**Pull request merged**